docs/context/compaction.md

```diff
@@ -0,0 +1,63 @@
+# Compress workflow context for performance
+
+As an ADK agent runs a workflow it builds up information, including user
+instructions, collected data, tool responses, and generated content. As the size
+of this in-process data grows, agent processing times typically also increase.
+More and more data is sent to the generative AI model used by the agent,
+increasing processing time and slowing down responses. The ADK Context
+Compaction feature is designed to reduce the size of this in-process data, or
+context, by summarizing older parts of the agent workflow event history. 
+
+The Context Compaction feature uses a *sliding window* approach for workflow
+event data. When configured, it summarizes data from older events once it
+reaches a threshold of a specific number of workflow events.
+
+## Configure context compaction
+
+Add context compaction to your agent workflow by adding an Event Compaction
+Configuration setting to the App object of your workflow. As part of the
+configuration, you must specify an AI model to perform the event summarization,
+as well as set a compaction interval and overlap size, as shown in the following
+sample code:
+
+```python
+from google.adk.apps.sliding_window_compactor import SlidingWindowCompactor
+from google.adk.apps.app import EventsCompactionConfig
+from google.adk.models import Gemini
+
+# Define AI model to create summaries.
+summarization_llm = Gemini(model="gemini-2.5-flash")
+my_compactor = SlidingWindowCompactor(llm=summarization_llm)
+
+app = App(
+    name='test',
+    root_agent=Mock(spec=BaseAgent),
+    events_compaction_config=EventsCompactionConfig(
+        compactor=my_compactor,
+        compaction_interval=5,  # Trigger compaction every 5 new invocations.
+        overlap_size=2          # Include last 2 invocations from the previous window.
+    ),
+)
+```
+
+Once configured, the ADK `Runner` automatically handles the compaction
+process in the background after each complete agent invocation event.
+
```

Comment:
do you think if we should talk about it will be compacted into EventAction.compaction: https://github.com/google/adk-python/blob/731bb9078d01359ae770719a8f5c003680ed9f3e/src/google/adk/events/event_actions.py#L97.

Then in later model call, the event action compaction(summary) will be used to cover those ranges that it summarized over.